### ISSUE TITLE
Question - Do too many VC's affect app performance? 

### ISSUE THREAD

**Issue by otymartin** [uias/Pageboy#143]
@msaps 
This is more of a theoretical Question. Is there some general guideline as to how many viewControllers is too many that it can start affecting the apps performance even creating lag in older devices?. Or is `UIPageViewController` built on the same magic that makes reusable collectionView cells so efficient?

Im just curious because I have a layout with 4 VC's in production and would like to add like 2 more but need clarification before I go crazy.

**Comment by msaps** [comment]
@otymartin erm I have no concrete data on how/when it will affect performance, nor is the sorcery within `UIPageViewController` in the public domain. But, `UIPageViewController` definitely does some sort of magic under the hood to optimise performance, it is not simply the view controllers stacked next to each other in a scroll view.

It may be an interesting thing to test - lots of heavy view controllers (like 100) and see how it performs. 

In short; you should be fine with 6 or any remotely normal use case. But I certainly might have to have a go at investigating this a bit more

**Comment by otymartin** [comment]
@msaps Ok good idea will just test but good that you mention 6 VC's shouldnt be a problem.

**Comment by otymartin** [comment]
@msaps 
I do the test here for anyone interested
[](https://github.com/otymartin/Dark-Magic.git)
